### ISSUE TITLE
Systemctl kill 'service@*' sends signals to same process

### ISSUE THREAD

**Issue by hiqsol** [systemd/systemd#4112]
### Submission type
- [x] Bug report
### systemd version the issue has been seen with

> 229
### Used distribution

> Ubuntu 16.04.1 LTS 4.4.0-36-generic #55-Ubuntu SMP Thu Aug 11 18:01:55 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
### Expected behaviour you didn't see

I have up and running 2 services: heppy-comnet@1 and heppy-comnet@2.
When I run:

``` sh
systemctl kill 'heppy-comnet@*'
```

I want systemctl to send signal to both services.
In this simplified case - it should kill and restart both services. (Actually I want to send other signals to handle them inside of my service).
### Unexpected behaviour you saw

When I run `systemctl kill 'heppy-comnet@*'` it sends both signals to heppy-comnet@2
I see it from the list of running processes and I see it in journal only one of services is restarted:

> heppyd[22846]: Exception SystemExit: SystemExit(None,) in <module 'threading' from '/usr/lib/python2.7/threading.pyc'> ignored
> systemd[1]: heppy-comnet@2.service: Service hold-off time over, scheduling restart.
> systemd[1]: Stopped hEPPy COMNET 2.
> systemd[1]: Started hEPPy COMNET 2.

**Comment by evverx** [comment]
> systemd version the issue has been seen with
> 229

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_

https://github.com/systemd/systemd/releases

I can't reproduce:

``` sh
# systemd from the master branch

bash-4.3# systemctl cat hola@ --no-pager
# /etc/systemd/system/hola@.service
[Service]
Type=simple
ExecStart=/bin/sh -x -c 'sleep %i'

bash-4.3# systemctl start hola@100
bash-4.3# systemctl start hola@150

bash-4.3# systemctl status hola@100 --no-pager
● hola@100.service
   Loaded: loaded (/etc/systemd/system/hola@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2016-09-08 14:52:38 UTC; 16s ago
 Main PID: 194 (sleep)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/system-hola.slice/hola@100.service
           └─194 sleep 100

bash-4.3# systemctl status hola@150 --no-pager
● hola@150.service
   Loaded: loaded (/etc/systemd/system/hola@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2016-09-08 14:52:45 UTC; 23s ago
 Main PID: 197 (sleep)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/system-hola.slice/hola@150.service
           └─197 sleep 150

bash-4.3# trace-cmd start -e signal_generate -f 'common_pid == 1'
/sys/kernel/tracing/events/signal/signal_generate

bash-4.3# systemctl kill 'hola@*'

bash-4.3# trace-cmd show
         systemd-1     [000] dN..   441.203572: signal_generate: sig=15 errno=0 code=0 comm=sleep pid=194 grp=1 res=0
         systemd-1     [000] dN..   441.304888: signal_generate: sig=15 errno=0 code=0 comm=sleep pid=197 grp=1 res=0

bash-4.3# systemctl status hola@100 --no-pager
● hola@100.service
   Loaded: loaded (/etc/systemd/system/hola@.service; static; vendor preset: enabled)
   Active: inactive (dead)

bash-4.3# systemctl status hola@150 --no-pager
● hola@150.service
   Loaded: loaded (/etc/systemd/system/hola@.service; static; vendor preset: enabled)
   Active: inactive (dead)
```

**Comment by hiqsol** [comment]
I've repeated your procedure. And received results that are different from yours.
Also note two processes per service :-/
And no `Tasks:` in my case. Is it something new?

Hope the issue is just already fixed in fresh versions. I'll check it up when I'll have it.

```
18:55:15 root@plex > systemctl start hola@15000                                                            /etc/systemd/system
18:55:24 root@plex > systemctl start hola@15002                                                            /etc/systemd/system

18:55:27 root@plex > systemctl status 'hola@*'                                                             /etc/systemd/system
● hola@15000.service
   Loaded: loaded (/etc/systemd/system/hola@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2016-09-08 18:55:24 EEST; 5s ago
 Main PID: 5628 (sh)
   CGroup: /system.slice/system-hola.slice/hola@15000.service
           ├─5628 /bin/sh -x -c sleep 15000
           └─5630 sleep 15000

Sep 08 18:55:24 plex systemd[1]: Started hola@15000.service.
Sep 08 18:55:24 plex sh[5628]: + sleep 15000

● hola@15002.service
   Loaded: loaded (/etc/systemd/system/hola@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2016-09-08 18:55:27 EEST; 2s ago
 Main PID: 5635 (sh)
   CGroup: /system.slice/system-hola.slice/hola@15002.service
           ├─5635 /bin/sh -x -c sleep 15002
           └─5637 sleep 15002

Sep 08 18:55:27 plex systemd[1]: Started hola@15002.service.
Sep 08 18:55:27 plex sh[5635]: + sleep 15002

18:55:30 root@plex > systemctl kill 'hola@*'                                                               /etc/systemd/system
Failed to kill unit hola@15000.service: Unit hola@15000.service is not loaded.

18:55:34 root@plex > systemctl status 'hola@*'                                                             /etc/systemd/system
● hola@15002.service
   Loaded: loaded (/etc/systemd/system/hola@.service; static; vendor preset: enabled)
   Active: active (running) since Thu 2016-09-08 18:55:27 EEST; 18s ago
 Main PID: 5635 (sh)
   CGroup: /system.slice/system-hola.slice/hola@15002.service
           ├─5635 /bin/sh -x -c sleep 15002
           └─5637 sleep 15002

Sep 08 18:55:27 plex systemd[1]: Started hola@15002.service.
Sep 08 18:55:27 plex sh[5635]: + sleep 15002
```

**Comment by evverx** [comment]
> Also note two processes per service :-/

I think `/bin/sh` points to `dash`

Try `ExecStart=/bin/bash -x -c 'sleep %i`

> And no Tasks: in my case.

Seems like https://anonscm.debian.org/cgit/pkg-systemd/systemd.git/commit/debian?id=fe4d9d3ba0502f8622a1139f109d070ff20f1d06

> Is it something new?

See 03a7b521e3ffb7f5d1

> I'll check it up when I'll have it.

Ok

**Comment by evverx** [comment]
This bug was fixed in 169d7fb684b39d
